src/App.js

```diff
@@ -72,10 +72,10 @@ class App extends React.Component {
         this.setState({results: number});
     }
 
-    updateSortCriteria(e) {
+    updateSortMethod(e) {
         if (e) {
-            const sortCriteria = e.target.value;
-            this.setState({sort: sortCriteria});
+            const sortName = e.target.value; // we may have move this 'if' depending on how the 'select' option reacts;
+            this.setState({sort: sortName});
```

Comment:
style nit: it might be simpler to lookup `sortMethods[sortCriteria]` directly here, then pass it down as a prop to child components where required.

Reply:
I tried to do this at one point, but had trouble passing the props so decided to do it this way! Instead, we are updating state with a reference and passing that reference as props. Is it better to minimise external dependancies and provide everything a Child Component needs via props?